src/include/network/lan/complex.rb

```diff
@@ -393,14 +446,23 @@ def handleOverview(_key, event)
           # As it could easily happen that all interfaces are set to "no" (and
           # default route is unrecheable in such case) this explicite setup was
           # added.
+          # FIXME: not implemented in network-ng
           LanItems.set_default_route = true
 
           return :add
         when :edit
-          if LanItems.IsCurrentConfigured
-            LanItems.SetItem
-
-            if LanItems.startmode == "managed"
+          iface = config.interfaces.find(iface_name)
+          # FIXME: turn following lines into something like builder.load(iface) ?
+          @builder.name = iface.name
+          @builder.type = iface.type
+          if iface.configured
+            @builder.load_sysconfig(iface.config)
+            @builder.load_s390_config(LanItems.s390_ReadQethConfig(iface.hardware.name)) if iface.hardware
+            # FIXME: move to more appropriate place
+            LanItems.operation = :edit
```

Comment:
I hope config itself can recognize if it already existed before or not. Maybe something trivial like system_name ( so name before all renames? ) and if it is nil, then it means new and of course nice method for it.

Reply:
yes ... already planed and described somewhere in this PR. This operation can be identified according to configuration name (I've introduced interface.name to be "persistent" and reflects current state of the system) name exists -> edit action, name do not exist -> add action